### ISSUE TITLE
[Failing Test]: Python 3.7 postcommit is red

### ISSUE THREAD

**Issue by tvalentyn** [apache/beam#26272]
### What happened?

Seeing 
```
test_1     |     raise BeamIOError("Match operation failed", exceptions)
test_1     | apache_beam.io.filesystem.BeamIOError: Match operation failed with exceptions {'azfs://devstoreaccount1/container/*': BeamIOError('List operation failed with exceptions {\'azfs://devstoreaccount1/container/\': HttpResponseError(\'The API version 2022-11-02 is not supported by Azurite. Please upgrade Azurite to latest version and retry. If you are using Azurite in Visual Studio, please check you have installed latest Visual Studio patch. Azurite command line parameter "--skipApiVersionCheck" or Visual Studio Code configuration "Skip Api Version Check" can skip this error. \\nRequestId:0b3deb47-2f6a-44ca-b1b6-1ae05f2090fb\\nTime:2023-04-13T07:23:16.447Z\\nErrorCode:InvalidHeaderValue\')}')}
test_1     | ERROR: InvocationError for command /app/sdks/python/target/.tox/azure_integration_test/bin/python -m apache_beam.examples.wordcount --input 'azfs://devstoreaccount1/container/*' --output azfs://devstoreaccount1/container/py-wordcount-integration --azure_connection_string 'DefaultEndpointsProtocol=https;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=https://azurite:10000/devstoreaccount1;' (exited with code 1)
test_1     | azure_integration_test run-test-post: commands[0] | bash /app/sdks/python/scripts/run_tox_cleanup.sh
test_1     | WARNING: test command found but not installed in testenv
test_1     |   cmd: /bin/bash
test_1     |   env: /app/sdks/python/target/.tox/azure_integration_test
test_1     | Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
test_1     | 
test_1     | DEPRECATION WARNING: this will be an error in tox 4 and above!
test_1     | ___________________________________ summary ____________________________________
test_1     | ERROR:   azure_integration_test: commands failed
azure_it-python_3_7-jenkins-beam_postcommit_python37-6557_test_1 exited with code 1
Stopping azure_it-python_3_7-jenkins-beam_postcommit_python37-6557_azurite_1 ... 
Stopping azure_it-python_3_7-jenkins-beam_postcommit_python37-6557_azurite_1 ... done
Aborting on container exit...

```

### Issue Failure

Failure: Test is continually failing

### Issue Priority

Priority: 1 (unhealthy code / failing or flaky postcommit so we cannot be sure the product is healthy)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by Abacn** [comment]
azureTest is back for a moment when Azurite 3.24.0 released, but failing again now not see a reason in log, it is just

```
azure_it-python_3_7-jenkins-beam_postcommit_python37-6784_test_1 exited with code 1
```

hdfs test also start failing:

```
hdfs_integration_test: failed with bash is not allowed, use allowlist_externals to allow it
```

**Comment by Abacn** [comment]
" failed with bash is not allowed, use allowlist_externals to allow it" is a tox error, likely due to #26082

**Comment by Abacn** [comment]
Let's close this as fix #27089 in. However the azure test still failing, use #27019 for tracking
